searchlib/src/vespa/searchlib/queryeval/children_iterators.h

```diff
@@ -0,0 +1,71 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "searchiterator.h"
+#include <assert.h>
+
+namespace search::queryeval {
+
+class MultiSearch;
+
+/**
+ * This class owns a set of SearchIterator instances, for
+ * holding them or passing ownership around.
+ **/
+class ChildrenIterators {
```

Comment:
I do not think 'children' needs to be part of the name. Maybe just SearchIteratorList or something similarly generic